### PR TITLE
docs: add release notes for 5.10.0

### PR DESCRIPTION
**Pull request by devcorpio**
- Release notes for 5.10.0

### CODE REVIEW

**Comment by devcorpio** [comment]
@elasticmachine run elasticsearch-ci/docs skiplinkcheck

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -29,6 +29,17 @@ See Conventional Commits (https://conventionalcommits.org) for commit guidelines
 [[release-notes-5.x]]
 === RUM JS Agent version 5.x
 
+[[release-notes-5.10.0]]
+==== 5.10.0 (2021-12-07)
+[float]
+===== Features
+* Make server prefix URL used to make requests to the APM Server configurable: {issue}1078[#1078]
```
Comment (vigneshshanmugam) [inline CHANGELOG.asciidoc:36]: So in release notes, its good to eloborate on a specific feature than keeping it short. I would write this as 

```suggestion
* Introduce new config option `serverUrlPrefix` to make the APM server URL configurable: {issue}1078[#1078]
```

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -29,6 +29,17 @@ See Conventional Commits (https://conventionalcommits.org) for commit guidelines
...
+* Expose Transaction, Span, TransactionOptions and SpanOptions types definition: {issue}1045[#1045] and {issue}995[#995]
+* Set to empty string span destination deprecated fields: {issue}1098[#1098]
```
Comment (vigneshshanmugam) [inline CHANGELOG.asciidoc:38]:
```suggestion
* Set the values to empty string for deprecated span destination metrics: {issue}1098[#1098]
```

**Review by vigneshshanmugam — APPROVED** [review]

**Pull request merged**